api/v1alpha1/temporalworker_webhook.go

```diff
@@ -73,21 +73,51 @@ func (r *TemporalWorkerDeployment) validateForUpdateOrCreate(ctx context.Context
 		return nil, apierrors.NewBadRequest("expected a TemporalWorkerDeployment")
 	}
 
+	return validateForUpdateOrCreate(nil, dep)
+}
+
+func validateForUpdateOrCreate(old, new *TemporalWorkerDeployment) (admission.Warnings, error) {
 	var allErrs field.ErrorList
 
-	if len(dep.Name) > maxTemporalWorkerDeploymentNameLen {
+	if len(new.GetName()) > maxTemporalWorkerDeploymentNameLen {
 		allErrs = append(allErrs,
-			field.Invalid(field.NewPath("name"), dep.Name, fmt.Sprintf("cannot be more than %d characters", maxTemporalWorkerDeploymentNameLen)),
+			field.Invalid(field.NewPath("metadata.name"), new.GetName(), fmt.Sprintf("cannot be more than %d characters", maxTemporalWorkerDeploymentNameLen)),
 		)
 	}
 
+	if new.Spec.RolloutStrategy.Strategy == UpdateProgressive {
```

Comment:
I wonder if the validation for strategies should live with the strategy code rather than living in the webhook admission system?

Reply:
I agree that would make a lot of sense, but unfortunately it's a little tricky to avoid import cycles if we want to import a validation function in the `v1alpha1` package. I refactored the rollout validation into a separate function for to make it easier to move it around in the future though.